### ISSUE TITLE
lost HP back to 100% after next log in

### ISSUE THREAD

**Issue by JustForVS** [anegostudios/VintageStory-Issues#2250]
Game Version: 1.17.8
Platform: Windows 10
Modded: Yes (hudclock-2.8.0, ProspectorInfo-4.1.1, singlepause-1.0.3, ZoomButton1.3.0)
SP/MP: Singleplayer

### Description
When leaving the game with less then 100% HP, HP are back at 100% after next log in.
I already realised this a long time ago but forgot about it. Not sure if it always happens, just made three tests in a row and it happened everytime. 

### How to reproduce
1. Get some damage.
2. Log out with less than 100% HP and log in again
3. HP are back at 100%

### Expected behavior
HP should be the same when logging out and reentering the game

### Screenshots
Attach a screenshot if it helps explain the issue / shows a visual bug.

### Logs
```
Paste crash between the triple backticks.

In the game's Logs folder you may be able to find more information
inside client-main.txt and server-main.txt. Alternatively, you may use
gist.github.com to upload logs, linking them here, if they're too large.
```

**Comment by JustForVS** [comment]
After some more tests it seems to need a minimum of HP left to get full HP back. With around 24 HP at 100%, HP were restored when loging out with at least around 16 HP left. When loging out with less, HP stay at current level after log in.

**Comment by wasakoki** [comment]
i had this issue on a vanilla dedicated server during the RCs. i couldn't reproduce it myself so i never make a bug report. deleting my playerdata "fixed" the issue so if i had to guess it's related to playerdata messing up somehow

**Comment by JustForVS** [comment]
Hm, after testing with a new game without success you seem right. But as I'm pretty sure I had this issue in my last game over a year ago and you also had it, it seems to be not so rare.

**Comment by Craluminum2413** [comment]
Closing due to inactivity

**Comment by JustForVS** [comment]
Unfortunately this still happens. 
New game started with v 1.18.something, currently v 1.18.5

**Comment by CookieBoy21** [comment]
Issue is still in game, you just need to have to eat something to have more hp. 

<img width="1280" alt="Vintagestory Screenshot 2024 09 26 - 16 40 06 74" src="https://github.com/user-attachments/assets/90dda1f7-9fa1-4e64-98e5-8b1e36bb185c">


https://github.com/user-attachments/assets/be19618c-ecf4-46b7-b447-2ba80de62ded


https://github.com/user-attachments/assets/84a2676b-88e8-41ef-8cdf-666dbd31585d

**Comment by Craluminum2413** [comment]
@CookieBoy21 Are you on 1.20?

**Comment by Craluminum2413** [comment]
I won't reopen it until someone reproduces issue in 1.20

**Comment by CookieBoy21** [comment]
I used v1.20 pre2. You can check it on video when I'm on main menu on bottom right.

**Comment by Th3Dilli** [comment]
unfortunately i was not able to reproduce it yet, if anyone is able to that would be really helpful

**Comment by CookieBoy21** [comment]
You must have more max hp like 20 for example. Then when you will have 80% of your hp save the game and load it again. In #4032 I wrote when this bug happen to me.

**Comment by CookieBoy21** [comment]
@Th3Dilli Did you tried to reproduce it again? If not yet, try fill 3 nutrition bars to max by eating food like grain, protein and vegetable.  After it, you should have like over 19 max hp. Now you can hurt yourself like 3-4, hp and then you can save the game and load it again. After it, you should have full hp. Also, I found out that using command for giving yourself more max hp don't work with it.
<img width="934" alt="Vintagestory Screenshot 2024 11 06 - 22 18 21 62" src="https://github.com/user-attachments/assets/31200382-437b-47e5-81ff-f3156d1e5ad4">


<img width="934" alt="Vintagestory Screenshot 2024 11 06 - 22 18 41 14" src="https://github.com/user-attachments/assets/fb26186a-934f-453a-bb13-a571d3239e12">

**Comment by radfast** [comment]
Thanks all who have contributed and kept this issue open.  We have finally fixed it in upcoming 1.20.0-rc.6
